### PR TITLE
CLOUDP-98311: Add tests for ops manager 5.0

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-98311

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

Screenshot 2021-08-23 at 15 07 19

### CODE REVIEW

File: build/ci/ego [hunk 1]
```diff
@@ -118,9 +118,6 @@ _determine_architecture() {
         elif grep -qE '^8' /etc/debian_version; then
             OSVER=8
 
-        elif grep -qE '^7' /etc/debian_version; then
```
Comment (gssbzn) [inline build/ci/ego:121]: remove deprecated OSes

File: build/ci/evergreen.yml [hunk 2]
```diff
@@ -902,8 +902,56 @@ tasks:
           MCLI_OPS_MANAGER_URL: ${mcli_ops_manager_url}
           MCLI_SERVICE: cloud-manager
           E2E_TAGS: cloudmanager,generic
-  - name: ops_manager_generic_4_4_e2e
-    tags: ["opsmanager", "general"]
+  # Deploy a replica set, we need to be careful on running parallel modifications to the automation config
+  - name: cloud_manager_deploy_replica_set_e2e
```
Comment (gssbzn) [inline build/ci/evergreen.yml:906]: couple of things in this file, grouping ops manager tests by ops manager version, adding OM5.0 removing OM4.2 (EOL)

File: build/ci/set-up-ops-manager.sh [hunk 3]
```diff
@@ -48,21 +48,33 @@ echo "generate password for owner"
...
 echo "generate email for owner"
-MCLI_PUBLIC_API_KEY=$(date +%s | sha256sum | base64 | head -c 8)@ops-manager-team
-export MCLI_PUBLIC_API_KEY
+email=$(date +%s | sha256sum | base64 | head -c 8)@ops-manager-team
+
 
 echo "create first user"
-MCLI_PRIVATE_API_KEY=$(./bin/mongocli om owner create --firstName evergreen --lastName evergreen --email "${MCLI_PUBLIC_API_KEY}" --password "${password}" --accessListIp "127.0.0.1/1" -o="go-template={{.APIKey}}")
-export MCLI_PRIVATE_API_KEY
+cat <<EOF > credentials.tmpl
```
Comment (gssbzn) [inline build/ci/set-up-ops-manager.sh:55]: changing to use programmatic API keys from personal API keys

**Review by fmenezes — APPROVED** [review]
Awesome 🥳

File: build/ci/set-up-ops-manager.sh [hunk 4]
```diff
@@ -48,21 +48,33 @@ echo "generate password for owner"
...
+#!/bin/bash
+
+set -euo pipefail
+
+export MCLI_PUBLIC_API_KEY={{.ProgrammaticAPIKey.PublicKey}}
+export MCLI_PRIVATE_API_KEY={{.ProgrammaticAPIKey.PrivateKey}}
```
Comment (andreaangiolillo) [inline build/ci/set-up-ops-manager.sh:61]: [q] I don't see where `.ProgrammaticAPIKey.PublicKey` is coming from

Comment (gssbzn) [reply build/ci/set-up-ops-manager.sh:61]: this is a template the command will output as (see line 64 `-o=go-template-file=`, the value will come from the api response https://docs.opsmanager.mongodb.com/current/reference/api/user-create-first/

Comment (andreaangiolillo) [reply build/ci/set-up-ops-manager.sh:61]: Thank you very much! I have missed how we were using this file. Great job on this!! 💯 🥇

**Pull request merged**